### DISCUSSION TITLE
DAI 2024-2025 - Practical work 2 - Mathieu Rabot and Florian Duruz

### DISCUSSION THREAD

**Discussion by Nomeos** [heig-vd-dai-course/heig-vd-dai-course#466] [Show and tell]
Ideas :

1. Tron
3. Bomberman (risky)
4. Pokemon (risky)

Currently learning how to use Lanterna to determine the targeted project.

**Comment by Cat-du-Fromage** [comment]
Mise à jour : 
Nous partons sur un jeu Bomberman mais en utilisant la librairie java.awt (graphic) et javax.swing (fenêtre), lanterna s'est révélé être un enfer pour créer un jeu.

**Reply by ludelafo** [reply]
OK pour nous, attention au temps à disposition.

**Comment by Cat-du-Fromage** [comment]
Question concernant le docker, ce dernier doit uniquement pouvoir lancer un serveur? et ce serveur peut-il être joueur aussi ou doit-il pouvoir être juste un "maître du jeu"?

**Comment by Cat-du-Fromage** [comment]
J'ai de gros soucis lié à l'utilisation de l'interface graphique pour l'image docker et apparemment wsl rend les chose très compliqué, est-ce que les images doivent obligatoirement afficher l'interface graphique pour avoir les points?

**Reply by ludelafo** [reply]
Bonjour Florian,

Uniquement le serveur est suffisant dans votre cas. Je sais qu'il est possible d'exécuter des applications graphiques avec Docker mais c'est bien trop avancé dans le contexte de DAI.

Avez-vous réussi à faire fonctionner la partie serveur ?

**Reply by Cat-du-Fromage** [reply]
Oui on peut lancer le serveur, cependant, je précise que le jeu n'est pas prévu pour fonctionner en "headless", le serveur doit aussi être un joueur(donc un Host) on peut donc lancer le docker en tant que serveur mais on ne pourra pas lancer le jeu car seul le serveur à accès au bouton "start" qui lance la partie.
En revanche les joueurs pourrons se mettre en "prêt" dans le menu principal

**Comment by Cat-du-Fromage** [comment]
lien vers le projet :
https://github.com/Nomeos/DAI-2024-2025-Practical-work-2-Mathieu-Rabot-and-Florian-Duruz
Hash du dernier commit:
67a661bbc328adb28320c341ab6f8b422167abb5
